portal-web/test/functional/com/liferay/portalweb/macros/Portlet.macro

```diff
@@ -49,7 +49,7 @@ definition {
 		while (!((IsElementPresent(locator1 = "Home#PORTLET")) || ("${i}" == "3"))) {
 			var i = ${i} + 1;
 
-			Pause(locator1 = "30000");
+			Pause(locator1 = "3000");
```

Comment:
I am very concerned about the stability of the tests with this change...hopefully it doesn't break that many tests. But I also think that it's ridiculous to wait 30 secs (x3) for a portlet to show up on tests. We'll see what the test results show.

Reply:
10x is indeed quite a big change. 😱

Reply:
I'd suggest this whole macro (and `addPGCmd`) to be looked into.

At the end, after all these manual steps, we're basically invoking `Liferay.Portlet.add` from https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/portlet.js#L155

That will trigger a `c/portal/update_layout` cmd (which can also probably be directly triggered avoiding all this navigation) and in the end it will fire a `addPortlet` event that can be used to detect when the portlet has been successfully added to the page.

So, I'd say:
- Listen to [`addPortlet`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/portlet.js#L195) should at least get rid of timing inconsistencies
- Figuring out the proper payload for `Liferay.Portlet.add(...)` could help avoid all the clicking & waiting complexity
- Directly triggering an `/c/portal/update_layout` command

Reply:
Thanks for the suggestions @jbalsas! I think listening to addPortlet would be easy to include for this PR, we'll test it out. I created https://issues.liferay.com/browse/LRQA-65669 to look further into the other two improvement points.

Reply:
Hi @jbalsas , I don't see the addPortlet getting triggered. Not sure if that's what we're expecting.
https://user-images.githubusercontent.com/35235266/110373242-23931b80-8004-11eb-8498-5070f7935dc8.mp4

But I see portletReady event is fired. We could maybe track that instead?
![PortletReady](https://user-images.githubusercontent.com/35235266/110373369-458c9e00-8004-11eb-932e-4db11291569a.png)

Reply:
oic...it's not fired for widgets added to ContentPage but it is fired for widgets added to a Widget Page. Did we want to include the addPortlet event when adding a portlet to Content Page too?

![WidgetPageAddPortletListener](https://user-images.githubusercontent.com/35235266/110373912-e1b6a500-8004-11eb-8950-b0462bcec32f.png)

Reply:
Hey @john-co, yeah, `portletReady` will be less implementation-specific since it's always triggered from https://github.com/liferay-frontend/liferay-portal/blob/master/portal-web/docroot/html/portal/render_portlet.jsp if I remember correctly.

> oic...it's not fired for widgets added to ContentPage but it is fired for widgets added to a Widget Page. Did we want to include the addPortlet event and others when adding a portlet to Content Page too?

Ideally these would be consistent, but they're also different methods handled by different apps, so there's no strong motivation to make them the same, specially if we already have some low-level common denominator.